src/authorize.js

```diff
@@ -1,12 +1,12 @@
 (function (global) {
 
-  function extractParams() {
+  function extractParams(url) {
     //FF already decodes the URL fragment in document.location.hash, so use this instead:
-    var location = RemoteStorage.Authorize.getLocation(),
-        hashPos  = location.href.indexOf('#'),
+    var location = url || RemoteStorage.Authorize.getLocation().href,
+        hashPos  = location.indexOf('#'),
```

Comment:
Yes, I saw the comment.
I'm not sure if it has anything to do with my changes though? It seems to me that this is functionally identical to the previous implementation when no `url` is passed. 

I might have missed your point.